src/cmd/tools/crane.go

```diff
@@ -82,3 +86,62 @@ func zarfCraneCatalog(cranePlatformOptions *[]crane.Option) *cobra.Command {
 
 	return craneCatalog
 }
+
+// Wrap the original crane list with a zarf specific version
+func zarfCraneList(cranePlatformOptions *[]crane.Option) *cobra.Command {
+	craneList := craneCmd.NewCmdList(cranePlatformOptions)
+
+	eg := `  # list the tags for a repo internal to Zarf
+  $ zarf tools registry ls 127.0.0.1:31999/stefanprodan/podinfo
+
+  # list the tags for a repo hosted at reg.example.com
+  $ zarf tools registry ls reg.example.com/stefanprodan/podinfo`
+
+	craneList.Example = eg
+	craneList.Args = nil
+
+	originalListFn := craneList.RunE
+
+	craneList.RunE = func(cmd *cobra.Command, args []string) error {
+		if len(args) < 1 {
+			message.Fatal(nil, lang.CmdToolsCraneListNoRepoSpecified)
+		}
+
+		// Try to connect to a Zarf initialized cluster otherwise then pass it down to crane.
+		zarfCluster, err := cluster.NewCluster()
+		if err != nil {
+			return originalListFn(cmd, args)
+		}
+
+		// Load the state
+		zarfState, err := zarfCluster.LoadZarfState()
+		if err != nil {
+			return err
+		}
+
+		// Check to see if it matches the existing internal address.
+		if !strings.HasPrefix(args[0], zarfState.RegistryInfo.Address) {
+			return originalListFn(cmd, args)
+		}
+
+		// Open a tunnel to the Zarf registry
+		tunnelReg, err := cluster.NewZarfTunnel()
+		if err != nil {
+			return err
+		}
+		err = tunnelReg.Connect(cluster.ZarfRegistry, false)
+		if err != nil {
+			return err
+		}
+
+		// Add the correct authentication to the crane command options
+		authOption := config.GetCraneAuthOption(zarfState.RegistryInfo.PullUsername, zarfState.RegistryInfo.PullPassword)
+		*cranePlatformOptions = append(*cranePlatformOptions, authOption)
+		registryEndpoint := tunnelReg.Endpoint()
+
+		return originalListFn(cmd, []string{strings.Replace(args[0],
+			fmt.Sprintf("%s/", zarfState.RegistryInfo.Address), fmt.Sprintf("%s/", registryEndpoint), 1)})
```

Comment:
```suggestion
		return originalListFn(cmd, args)
```
This can be simplified if the above is accepted.